src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -535,6 +535,21 @@ public Response searchDatasets(@Auth AuthUser authUser, @QueryParam("query") Str
     }
   }
 
+  @POST
+  @Path("/search/index")
+  @Consumes("application/json")
+  @Produces("application/json")
+  public Response searchDatasetIndex(@Auth AuthUser authUser, String query) {
+    try {
+      User user = userService.findUserByEmail(authUser.getEmail());
+      // Use the searchDataset method as a dummy method to validate the query
+      List<Dataset> datasets = datasetService.searchDatasets(query, user);
+      return Response.ok().entity(unmarshal(datasets)).build();
```

Comment:
I think maybe there's a misunderstanding of the required model. We're not returning real datasets here. We're returning a mock of what the elastic search query will return which is not a dataset, but instead is a term object that looks a lot like a dataset, but is different (see [DUOS-2469](https://broadworkbench.atlassian.net/browse/DUOS-2469) for more info). For this PR, it could look like this:
```
{
    datasetId: integer
    datasetIdentifier: string
    description: string
    studyName: string
    studyId: integer
    participantCount: integer
    phenotype: string
    species: string
    piName: string
    dataSubmitter: string
    dataCustodian: string
    dataUse: [list of strings (data use abbreviations)]
    dataTypes: [list of strings]
    dataLocation: string
    dacName: string
    openAccess: boolean
    approvedUsers: [list of strings (usernames? userids?)]
}
```